Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -519,22 +522,35 @@ createDriverFee merchantId merchantOpCityId driverId rideFare currency newFarePa
       _ -> return (0, 0, 0, False)
     let totalDriverFee = govtCharges + platformFee + cgst + sgst
-    mbDriverPlan <- getPlanAndPushToDefualtIfEligible transporterConfig freeTrialDaysLeft isSpecialZoneCharge
     now <- getLocalCurrentTime transporterConfig.timeDiffFromUtc
-    lastDriverFee <- QDF.findLatestFeeByDriverIdAndServiceName driverId serviceName
-    driverFee <- mkDriverFee serviceName now Nothing Nothing merchantId driverId rideFare govtCharges platformFee cgst sgst currency transporterConfig (Just booking) isSpecialZoneCharge
     vehicle <- QV.findById driverId
+    let currentVehicleCategory = vehicle >>= (.category)
+    subscriptionConfig <- CQSC.findSubscriptionConfigsByMerchantOpCityIdAndServiceName merchantOpCityId serviceName
+    (mbDriverPlan, isOnFreeTrial) <- getPlanAndPushToDefualtIfEligible transporterConfig subscriptionConfig freeTrialDaysLeft' isSpecialZoneCharge
+    driverFee <- mkDriverFee serviceName now Nothing Nothing merchantId driverId rideFare govtCharges platformFee cgst sgst currency transporterConfig (Just booking) isSpecialZoneCharge currentVehicleCategory
+    lastElderSiblingDriverFee <- QDF.findLatestFeeByDriverIdAndServiceName driverId serviceName merchantOpCityId driverFee.startTime driverFee.endTime
+    restSiblingDriverFee <- do
+      case lastElderSiblingDriverFee of
+        Just lESDriverFee -> do
+          if lESDriverFee.hasSibling == Just True
+            then QDF.findAllChildsOFDriverFee merchantOpCityId driverFee.startTime driverFee.endTime DF.ONGOING serviceName [lESDriverFee.id]
+            else return []
+        Nothing -> return []
+    let lastDriverFee = DL.find (\dfee -> dfee.vehicleCategory == currentVehicleCategory) (restSiblingDriverFee <> catMaybes [lastElderSiblingDriverFee])
     let isEnableForVariant = maybe True (`elem` transporterConfig.variantsToEnableForSubscription) (vehicle <&> (.variant))
     let toUpdateOrCreateDriverfee = (totalDriverFee > 0 || (totalDriverFee <= 0 && transporterConfig.isPlanMandatory && isJust mbDriverPlan)) && isEnableForVariant
-    when (toUpdateOrCreateDriverfee && isEligibleForCharge transporterConfig freeTrialDaysLeft isSpecialZoneCharge) $ do
+    when (toUpdateOrCreateDriverfee && isEligibleForCharge transporterConfig isOnFreeTrial isSpecialZoneCharge) $ do
       numRides <- case lastDriverFee of
         Just ldFee ->
           if now >= ldFee.startTime && now < ldFee.endTime
             then do
               QDF.updateFee ldFee.id rideFare govtCharges platformFee cgst sgst now True booking isSpecialZoneCharge
               return (ldFee.numRides + 1)
             else do
-              QDF.create driverFee
+              let elderSiblingId = if (lastElderSiblingDriverFee <&> (.id)) == Just ldFee.id then Nothing else lastElderSiblingDriverFee <&> (.id)
+              whenJust elderSiblingId $ \elderSiblingId' -> QDF.updateHasSiblingInDriverFee elderSiblingId'
```

Comment:
Same should be done in case `lastDriverFee` is Nothing?

Reply:
Yes, missed here fixing it.